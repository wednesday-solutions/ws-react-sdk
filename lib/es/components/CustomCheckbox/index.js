import{_ as o,a as e}from"../_rollupPluginBabelHelpers-c2a46d82.js";import"antd/lib/checkbox/style";import r from"antd/lib/checkbox";import"../_rollup-plugin-inject-process-env-1259e43d.js";import c from"react";import{c as n}from"../colors-93a064d8.js";import t from"styled-components";var a,l=t(r)(a||(a=o(["\n  .ant-checkbox .ant-checkbox-inner {\n    border-color: ",";\n  }\n  .ant-checkbox-checked .ant-checkbox-inner {\n    background-color: ",";\n  }\n"])),n.pawlyBlue,n.pawlyBlue);function i(o){return c.createElement(l,e({"data-testid":"custom-checkbox"},o))}export default i;
