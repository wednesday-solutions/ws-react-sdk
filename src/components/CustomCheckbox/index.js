/**
 *
 * CustomCheckbox
 *
 */

import React from 'react';
import { Checkbox } from 'antd';
import colors from '@themes/colors';
import styled from 'styled-components';

const CB = styled(Checkbox)`
  .ant-checkbox .ant-checkbox-inner {
    border-color: ${colors.pawlyBlue};
  }
  .ant-checkbox-checked .ant-checkbox-inner {
    background-color: ${colors.pawlyBlue};
  }
`;

function CustomCheckbox(props) {
  return <CB data-testid={'custom-checkbox'} {...props} />;
}

export default CustomCheckbox;
