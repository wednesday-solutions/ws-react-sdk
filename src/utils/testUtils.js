import React from 'react';
import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react';
import { DEFAULT_LOCALE, translationMessages } from '@src/i18n';
import IntlGlobalProvider from '@components/IntlGlobalProvider';

export const renderWithIntl = children => {
  return render(
    <IntlProvider locale={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <IntlGlobalProvider>{children}</IntlGlobalProvider>
    </IntlProvider>
  );
};

export const getComponentStyles = (Component, props = {}) => {
  renderWithIntl(Component(props));
  const { styledComponentId } = Component(props).type;
  const componentRoots = document.getElementsByClassName(styledComponentId);
  // eslint-disable-next-line no-underscore-dangle
  return window.getComputedStyle(componentRoots[0])._values;
};

export const timeout = ms => new Promise(resolve => setTimeout(resolve, ms));
export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
});

export default {
  timeout,
  renderWithIntl,
  getComponentStyles,
  apiResponseGenerator
};
