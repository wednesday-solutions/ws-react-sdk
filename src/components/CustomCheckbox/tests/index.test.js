/**
 *
 * Tests for CustomCheckbox
 *
 */

import React from 'react';
import { renderWithIntl } from '@utils/testUtils';
import CustomCheckbox from '../index';

describe('<CustomCheckbox />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<CustomCheckbox />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 CustomCheckbox component', () => {
    const { getAllByTestId } = renderWithIntl(<CustomCheckbox />);
    expect(getAllByTestId('custom-checkbox').length).toBe(1);
  });
});
