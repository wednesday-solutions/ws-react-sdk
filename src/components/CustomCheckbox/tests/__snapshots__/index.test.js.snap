// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomCheckbox /> should render and match the snapshot 1`] = `
<body>
  <div>
    <label
      class="ant-checkbox-wrapper sc-bdfBwQ jXNEbi"
    >
      <span
        class="ant-checkbox"
      >
        <input
          class="ant-checkbox-input"
          data-testid="custom-checkbox"
          type="checkbox"
          value=""
        />
        <span
          class="ant-checkbox-inner"
        />
      </span>
    </label>
  </div>
</body>
`;
